src/main/java/org/broadinstitute/dsm/route/PatchRoute.java

```diff
@@ -537,4 +547,19 @@ private NameValue setAdditionalValue(String additionalValue, @NonNull Patch patc
             throw new RuntimeException("DBElement not found in ColumnNameMap: " + additionalValue);
         }
     }
+
+    private void triggerParticipantEvent(DDPInstance ddpInstance, Patch patch, Value action){
+        inTransaction((conn) -> {
+            Optional<EventTypeDto> eventType = EventTypeDao.getEventTypeByEventTypeAndInstanceId(conn, action.getName(), ddpInstance.getDdpInstanceId());
+            if (eventType != null && !eventType.isEmpty()) {
+                if (!EventDao.hasTriggeredEventByEventTypeAndDdpParticipantId(conn, action.getName(), patch.getParentId())) {
+                    EventUtil.triggerDDP(conn, eventType, patch.getParentId());
```

Comment:
if this line was the key point of having connection passed to `getEventTypeByEventTypeAndInstanceId` just to avoid several connection, if think it still worth to eliminate connection from `getEventTypeByEventTypeAndInstanceId`, `EventUtil` needs to be refactored later, since, I think it does more than being utility class only.